### DISCUSSION TITLE
Choosing Laravel for our new project

### DISCUSSION THREAD

**Discussion by withgem** [community/community#130395] [Programming Help] [answered]
### Body

Our team has limited experience with Laravel. While we recognize it as a highly demanded framework, our research has revealed some concerns regarding its performance and architecture. Therefore, we seek reliable insights from experienced developers who have worked extensively with Laravel and other PHP frameworks. Your expertise will be invaluable in guiding our decision-making process.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by DarioChiappello** [comment]
If you are looking to develop systems based on microservices architecture, you can use Lumen (it is a Laravel microframework oriented to the microservices development) or you can also use Symfony. If you are looking for monolithic application development, Laravel is perfect. But it depends on the case

**Comment by withgem** [comment]
As @DarioChiappello mentioned, if you are using microservices, Laravel might not be the best choice, especially if you are using event-driven architecture instead of HTTP API calls for communication. Laravel is a heavy framework with many built-in functionalities and third-party services, which could lead to performance issues and maintenance burdens. Lumen cannot be recommended as it is no longer actively maintained (see the documentation). Symfony can perform slightly faster than Laravel, but the difference is minimal. If your services don't require many functionalities like authentication, queues, or other built-in features, raw PHP might be the best option.

For building a monolithic system with many feature demands, Laravel can be an excellent choice. However, if your system doesn't need those functionalities and you prefer more flexibility in your project, Symfony or raw PHP might be more suitable.

Conclusion :) It all depends on your project requirements. Conduct a thorough analysis of each technology and choose the one that best matches your needs!

**Comment by Lakshan-Madushanka** [comment, marked as answer]
Laravel is indeed a popular and widely used PHP framework, known for its elegant syntax, rich feature set, and strong community support. However, like any technology, it comes with both strengths and potential challenges, especially depending on your team's experience and the specific needs of your project.

Strengths of Laravel:

Developer Experience: Laravel's syntax is clean and expressive, which can significantly speed up development, especially for teams familiar with PHP. Its built-in tools like Artisan CLI, Eloquent ORM, and Blade templating engine simplify many common tasks.

Community and Ecosystem: Laravel has a large, active community that continually contributes packages, tutorials, and documentation. The Laravel ecosystem, including tools like Forge, Vapor, and Nova, provides a comprehensive suite of solutions for common development needs.

Scalability: While Laravel may not be as lightweight as some micro-frameworks, it is scalable with the right architecture and infrastructure. Tools like Horizon for queue management and Octane for performance optimization can help in scaling your applications.

Considerations:

Performance: Laravel is not the fastest PHP framework out-of-the-box due to its rich feature set. For high-performance requirements, careful optimization and caching strategies are necessary. You might need to consider alternatives like Lumen (Laravel’s micro-framework) for performance-critical applications.

Learning Curve: If your team has limited experience with Laravel, there might be a learning curve, particularly with mastering the conventions and leveraging Laravel’s full potential. However, the investment in learning can pay off in the long run.

Architecture: Laravel follows the MVC (Model-View-Controller) architecture, which works well for many projects. However, for very large applications, you may need to implement additional architectural patterns (such as Service Pattern, Repository Pattern, etc.) to maintain code organization and scalability.

Conclusion:
If your project aligns with Laravel's strengths—such as rapid development, a need for a rich feature set, and strong community support—it could be a great choice. However, if performance is a critical concern or your team is more comfortable with another PHP framework, those factors should weigh into your decision.

I’d recommend a pilot project or a proof of concept to gauge Laravel's fit for your team. This approach allows you to evaluate Laravel's performance and developer experience firsthand before making a larger commitment.
